polkadot/runtime/parachains/src/hrmp.rs

```diff
@@ -817,15 +961,19 @@ impl<T: Config> Pallet<T> {
 				"can't be `None` due to the invariant that the list contains the same items as the set; qed",
 			);
 
+			let system_channel = request.sender_deposit.is_zero();
```

Comment:
you could use is_system method of sender and recipient to determine if it system